tools/shared/src/main/scala/org/scalajs/core/tools/javascript/Printers.scala

```diff
@@ -263,138 +345,172 @@ object Printers {
-        case Null() =>
-          print("null")
+        case _: Null =>
+          printString("null")
 
-        case BooleanLiteral(value) =>
-          print(if (value) "true" else "false")
+        case tree: BooleanLiteral =>
+          printString(if (tree.value) "true" else "false")
 
-        case IntLiteral(value) =>
-          if (value >= 0)
-            print(value)
-          else
-            print("(", value, ")")
+        case tree: IntLiteral =>
+          val value = tree.value
+          if (value >= 0) {
+            printOne(value)
+          } else {
+            printChar('(')
+            printOne(value)
+            printChar(')')
+          }
 
-        case DoubleLiteral(value) =>
+        case tree: DoubleLiteral =>
+          val value = tree.value
           if (value == 0 && 1 / value < 0)
-            print("(-0)")
+            printString("(-0)")
           else if (value >= 0)
-            print(value)
-          else
-            print("(", value, ")")
+            printOne(value)
+          else {
+            printChar('(')
+            printOne(value)
+            printChar(')')
+          }
 
-        case StringLiteral(value) =>
-          print("\"", escapeJS(value), "\"")
+        case tree: StringLiteral =>
+          val value = tree.value
+          printString("\"")
+          printString(escapeJS(value))
+          printString("\"")
 
         // Atomic expressions
 
-        case VarRef(ident) =>
-          print(ident)
+        case tree: VarRef =>
+          printOne(tree.ident)
 
-        case This() =>
-          print("this")
+        case _: This =>
+          printString("this")
 
-        case Function(args, body) =>
-          print("(function")
-          printSig(args)
-          printBlock(body)
-          print(")")
+        case tree: Function =>
+          printString("(function")
+          printSig(tree.args)
+          printBlock(tree.body)
+          printChar(')')
 
         // Named function definition
 
-        case FunctionDef(name, args, body) =>
-          print("function ", name)
-          printSig(args)
-          printBlock(body)
+        case tree: FunctionDef =>
+          printString("function ")
+          printOne(tree.name)
+          printSig(tree.args)
+          printBlock(tree.body)
 
         // ECMAScript 6 classes
 
-        case ClassDef(optClassName, optParentClass, members) =>
-          print("class")
-          for (className <- optClassName)
-            print(" ", className)
-          for (parentClass <- optParentClass)
-            print(" extends ", parentClass)
-          print(" {"); indent
-          for (member <- members) {
+        case tree: ClassDef =>
+          printString("class")
+          for (className <- tree.className) {
+            printChar(' ')
+            printOne(className)
+          }
+          for (parentClass <- tree.parentClass) {
+            printString(" extends ")
+            printOne(parentClass)
+          }
```

Comment:
I guess `ClassDef` is relatively rare, but ... `.isDefined` and `.get`, while we're at it?